WorkloadManagementSystem/Utilities/RabbitMQSynchronizer.py

```diff
@@ -0,0 +1,107 @@
+'''RabbitMQSynchronizer
+
+  RabbitMQSynchronizer keeps the RabbitMQ user database  synchronized with the CS
+  RabbitMQ user database is updated according to changes in CS.
+'''
+
+from DIRAC import S_OK
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import hostHasProperties
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getDNsInGroup
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getDNForHost
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getHosts
+from DIRAC.Core.Utilities.RabbitMQAdmin import getAllUsers, deleteUsers
+from DIRAC.Core.Utilities.RabbitMQAdmin import setUsersPermissions, addUsersWithoutPasswords
+
+class RabbitMQSynchronizer(object):
+
+  def __init__( self ):
+
+    # Warm up local CS
+    # I am not sure whether it is needed but
+    # it was used in DIRAC.ResourceStatusSystem.Utilities.Synchronizer
+    warmUp()
+    self._accessUserGroup = 'lhcb_pilot'
+    self._accessProperty = 'GenericPilot'
+
+  def sync( self, _eventName, _params ):
+    '''
+    examples:
+      >>> s.sync( None, None )
+          S_OK()
+
+    :Parameters:
+      **_eventName** - any
+        this parameter is ignored, but needed by caller function.
+      **_params** - any
+        this parameter is ignored, but needed by caller function.
+    :return: S_OK
+    '''
+
+    valid_users = getDNsInGroup(self._accessUserGroup)
+    valid_hosts = getDNsForValidHosts(self._accessProperty)
+    updateRabbitMQDatabase(valid_users + valid_hosts)
+    return S_OK()
+
+def getDNsForValidHosts(accessProperty):
+  """Returns DN of hosts which contains accessProperty based on current
+     CS settings.
+  Args:
+    accessProperty(str):
+  Returns:
+    list: of hosts with accessProperty set.
+  """
+  retVal = getHosts()
+  if not retVal[ 'OK' ]:
+    return []
+  hosts = retVal['Value']
+  DNs = []
+  for host in hosts:
+    if hostHasProperties(host, [accessProperty]):
+      retVal = getDNForHost(host)
+      if not retVal[ 'OK' ]:
+        return []
```

Comment:
Right. i changed the logic. Now in case of error for given host, it just prints the message but tries with others anyway